mjidris/irondb
# Us 232

## Objective
Persisting user logins across all page components within the scope of the single page application and persistent logins upon closing browser.

## Detailed Description
Project hierarchy was changed. Routing is stored in Main.js and App.js now to facilitate a better hierarchy which allows for state context's producer to be at the top level which allows for all lower level consumers to have access to that data and instantly re-render the page when the state context variable is changed. When merging new builds please note that routing should no longer be placed in the index.js file!


## Screenshot (if applicable)

![image](https://user-images.githubusercontent.com/48575921/74310916-42a97600-4d3c-11ea-829e-f751112f1b0d.png)
![image](https://user-images.githubusercontent.com/48575921/74310926-4ccb7480-4d3c-11ea-85fb-1e6f6ec64fd0.png)

## How to Test (if applicable)

Install js-cookie library.
`npm install js-cookie --save`

I had an issue where my dependancies broke after I tried to start up the client again. I resolved this by updating my npm, removing all my dependancies and reinstalling. If you're unable to "npm start" try this:
```
npm install -g npm@latest
rm -rf node_modules
npm install
```
Afterwards you should be able to "npm start"

1. Login with any user credentials.
2. Check that navbar has changed to logged in state. 
3. Click around one navbar to ensure that login is persistent across all single page application components
4. Observe that help page shows additional information (data entry tutorial) while logged in.
5. Refresh browser to observe that login is persistent
6. Click logout to ensure that cookie and user session is wiped immediately. Page should seamless logout.

## Tests (if added)


## Requirements
[Taiga User Story](https://tree.taiga.io/project/cbrobsto-iron-db/us/232)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/234)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/235)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/236)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/238)
[Taiga Task](https://tree.taiga.io/project/cbrobsto-iron-db/task/239)
